cfgmgr/coppmgr.cpp

```diff
@@ -474,8 +548,54 @@ void CoppMgr::doCoppTrapTask(Consumer &consumer)
                 (trap_group == m_coppTrapConfMap[key].trap_group) &&
                 (trap_ids == m_coppTrapConfMap[key].trap_ids))
             {
-                it = consumer.m_toSync.erase(it);
-                continue;
+                std::vector<std::string> feature_keys;
+                m_cfgFeatureTable.getKeys(feature_keys);
+                std::vector<FieldValueTuple> feature_fvs;
+                /* duplicate check for the always_enabled field */
+                if (!m_coppTrapConfMap[key].is_always_enabled.empty() && (is_always_enabled == m_coppTrapConfMap[key].is_always_enabled))
+                {
+                    it = consumer.m_toSync.erase(it);
+                    continue;
+                }
+                else if (is_always_enabled == "true")
+                {
+                    /* if the value was changed from false to true,
+                    if the trap is not installed, install it.
+                    otherwise, do nothing. */
+
+                    m_coppAlwaysEnabledTraps.insert(key);
+
+                    if (m_coppTrapConfMap.find(key) == m_coppTrapConfMap.end())
+                    {
+                        string trap_group_trap_ids;
+                        addTrap(fvs, trap_ids, trap_group, is_always_enabled, trap_group_trap_ids);
+                    }
+                }
+                else
+                {
+                    /* if the value was changed from true to false,
+                    check if there is a feature enabled.
+                    if no, remove the trap. is yes, do nothing. */
+
+                    m_coppAlwaysEnabledTraps.erase(key);
+
+                    if (std::find(feature_keys.begin(), feature_keys.end(), key) != feature_keys.end())
+                    {
+                        m_cfgFeatureTable.get(key, feature_fvs);
```

Comment:
Can we cache the featureTable in a local datastructure in coppmgr class? Since we are listening to feature table notifications and we are reading feature table during init, it should be easier to build a cache.

Reply:
Done